src/utils.ts

```diff
@@ -258,8 +262,7 @@ export function executeLegacyOperation(
       const optionsIndex = args.length - 2;
       args[optionsIndex] = Object.assign({}, args[optionsIndex], { session: session });
     } else if (opOptions.session && opOptions.session.hasEnded) {
-      // TODO(NODE-3405): Replace this with MongoExpiredSessionError
-      throw new MongoDriverError('Use of expired sessions is not permitted');
+      throw new MongoExpiredSessionError('Use of expired sessions is not permitted');
```

Comment:
Let's make this error message the same has the one in applySession in sessions.ts
(maybe even make that the default message for expired session errors.)